sdk/core/System.ClientModel/src/Convenience/AsyncEnumerableResultOfT.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.ClientModel;
+
+#pragma warning disable CS1591 // public XML comments
+public abstract class AsyncEnumerableResult<T> : ClientResult, IAsyncEnumerable<T>, IAsyncDisposable
+    where T : IPersistableModel<T>
+{
+    protected internal AsyncEnumerableResult(PipelineResponse response) : base(response)
+    {
+    }
+
+    public abstract IAsyncEnumerator<T> GetAsyncEnumerator(CancellationToken cancellationToken = default);
+
+    protected abstract ValueTask DisposeAsyncCore();
+
+    protected abstract void Dispose(bool disposing);
```

Comment:
what is this for? i.e. who call it? BTW, FDGs (that Jeremy wrote) say to implement both IAsyncDisposabel and IDisposable

Reply:
Removed per discussion about moving disposal of resources into the enumerator.